calico-vpp-agent/services/service_handler.go

```diff
@@ -104,26 +108,47 @@ func (s *Server) updateCnatEntry(key string, entry *types.CnatTranslateEntry) (e
 	return nil
 }
 
-func (s *Server) advertiseSpecificRouteForExternalIP(extIP net.IP, withdraw bool) {
-	_, serviceExternalIPs, _ := s.getServiceIPs()
+func (s *Server) advertiseSpecificRoute(IPAddress net.IP, withdraw bool) {
```

Comment:
It might be worth adding a function that returns whether a list of IPBlocks contains an address or not.
this function could then be simplified as :
````
_, serviceBlocks, serviceLBBlocks := s.getServiceIPs()
if blockContains(serviceBlocks, addr) || blockContains(serviceLBBlocksm addr) {
    common.SendEvent (...)
}
````
Otherwise we're a bit under the impression this is a loop over all externalIPs